integration-tests/bats/profile.bats

```diff
@@ -0,0 +1,315 @@
+    dolt profile add --use-db defaultDB defaultTest
+
+    run dolt --profile nonExistentProfile sql -q "select * from altDB_tbl"
+    [ "$status" -eq 1 ] || false
+    [[ "$output" =~ "Failure to parse arguments: profile nonExistentProfile not found" ]] || false
+}
+
+@test "profile: additional flag gets used" {
+    cd altDB
+    dolt sql -q "create table test (pk int primary key)"
+    dolt sql -q "insert into test values (999)"
+    dolt add test
+    dolt commit -m "insert initial value into test"
+    cd ..
+
+    dolt profile add --user dolt --password "" userProfile
+    run dolt --profile userProfile --use-db altDB sql -q "select * from test"
+    [ "$status" -eq 0 ] || false
+    [[ "$output" =~ "999" ]] || false
+}
+
+@test "profile: duplicate flag overrides correctly" {
+    cd altDB
+    dolt sql -q "create table test (pk int primary key)"
+    dolt sql -q "insert into test values (999)"
+    dolt add test
+    dolt commit -m "insert initial value into test"
+    cd ..
+
+    dolt profile add --use-db defaultDB defaultTest
+    run dolt --profile defaultTest --use-db altDB sql -q "select * from test"
+    [ "$status" -eq 0 ] || false
+    [[ "$output" =~ "999" ]] || false
+}
+
+@test "profile: duplicate flag with non-duplicate flags in profile overrides correctly" {
+    cd altDB
+    dolt sql -q "create table test (pk int primary key)"
+    dolt sql -q "insert into test values (999)"
+    dolt add test
+    dolt commit -m "insert initial value into test"
+    cd ..
+
+    start_sql_server altDb
+    dolt --user dolt --password "" sql -q "CREATE USER 'steph' IDENTIFIED BY 'pass'; GRANT ALL PRIVILEGES ON altDB.* TO 'steph' WITH GRANT OPTION;";
+    dolt profile add --user "not-steph" --password "pass" --use-db altDB userWithDBProfile
+
+    run dolt --profile userWithDBProfile --user steph sql -q "select * from test"
+    [ "$status" -eq 0 ] || false
+    [[ "$output" =~ "999" ]] || false
+}
+
+@test "profile: duplicate flag with non-duplicate flags overrides correctly" {
+    cd altDB
+    dolt sql -q "create table test (pk int primary key)"
+    dolt sql -q "insert into test values (999)"
+    dolt add test
+    dolt commit -m "insert initial value into test"
+    cd ..
+
+    start_sql_server altDb
+    dolt --user dolt --password "" sql -q "CREATE USER 'steph' IDENTIFIED BY 'pass'; GRANT ALL PRIVILEGES ON altDB.* TO 'steph' WITH GRANT OPTION;";
+    dolt profile add --user "not-steph" --password "pass" userProfile
+
+    run dolt --profile userProfile --user steph --use-db altDB sql -q "select * from test"
+    [ "$status" -eq 0 ] || false
+    [[ "$output" =~ "999" ]] || false
+}
+
+@test "profile: dolt profile add adds a profile" {
+    run dolt profile add --use-db altDB altTest
+    [ "$status" -eq 0 ] || false
+
+    run dolt profile
+    [ "$status" -eq 0 ] || false
+    [[ "$output" =~ "altTest" ]] || false
+    [[ "$output" =~ "altDB" ]] || false
+}
+
+@test "profile: dolt profile add overwrites an existing profile" {
```

Comment:
I believe we shouldn't allow this. Let's just throw an error if they attempt to do this. They can delete and create if they really want to overwrite.